src/v/cluster/partition.h

```diff
@@ -321,6 +321,10 @@ class partition {
     ss::future<std::vector<rm_stm::tx_range>>
     aborted_transactions_cloud(const cloud_storage::offset_range& offsets);
 
+    model::producer_id highest_producer_id() {
```

Comment:
perhaps good to document implications of max_collectible_offset / sync() as a follow on.

Reply:
Will do in a follow up (short term, likely just a comment update)